pkgs/applications/networking/cluster/atlantis/default.nix

```
@@ -2,16 +2,20 @@
 
 buildGoModule rec {
   pname = "atlantis";
-  version = "0.22.3";
+  version = "0.27.0";
 
   src = fetchFromGitHub {
     owner = "runatlantis";
     repo = "atlantis";
     rev = "v${version}";
-    sha256 = "sha256-A/FT9t5Z+Iw1mVwS3d5Cc86A9e6jVbEtmEWroVUhhtw=";
+    hash = "sha256-a+xrmEHkSh5kicxIIxnoXgF9ep2ay5kCXwMR2sAVJIA=";
   };
 
-  vendorHash = "sha256-KUkh5yx+v5g0N4yIpgpt3i+uCtOtR0Jvf2PFQcGWtm8=";
+  patchPhase = ''
+    substituteInPlace main.go --replace dev ${version}
+  '';
```

Can you set using it with Go flags instead?
```suggestion
  ldflags = [ "-X=main.version=${version}" ];
```

BTW, we probably also should set the `commit` and `date` (with Unix epoch) flags: https://github.com/runatlantis/atlantis/blob/main/main.go#L30-L32

---

https://github.com/runatlantis/atlantis/blob/cea2db5fb0690b91f85f5da8cb0113a2cfac0e5c/Dockerfile#L40

It definitely seems to work, so we should use it instead of fragile `substituteInPlace` hacks.

---

Adding date will make this impure. Also, fetchFromGithub doesn't give you any information about the commit ID. Hardcoding the commit ID or switching to fetchgit just to add this output to the version subcommand is excessive.

---

> Adding date will make this impure.

This is why I suggested to use the Unix Epoch: https://en.wikipedia.org/wiki/Unix_time, not the actual build time.

> Also, fetchFromGithub doesn't give you any information about the commit ID. Hardcoding the commit ID or switching to fetchgit just to add this output to the version subcommand is excessive.

This is fair enough.

---

I've added the date in the RFC3339 format as recommended [here](https://goreleaser.com/cookbooks/using-main.version/).